pkg/store/bucket.go

```diff
@@ -474,12 +477,19 @@ func (s *BucketStore) TimeRange() (mint, maxt int64) {
 // Info implements the storepb.StoreServer interface.
 func (s *BucketStore) Info(context.Context, *storepb.InfoRequest) (*storepb.InfoResponse, error) {
 	mint, maxt := s.TimeRange()
-	// Store nodes hold global data and thus have no labels.
-	return &storepb.InfoResponse{
+
+	infoResp := &storepb.InfoResponse{
 		StoreType: component.Store.ToProto(),
 		MinTime:   mint,
 		MaxTime:   maxt,
-	}, nil
+		Labels:    make([]storepb.Label, 0, len(s.selectorLabels)),
+	}
+
+	for _, lset := range s.selectorLabels {
+		infoResp.Labels = append(infoResp.Labels, storepb.Label{Name: lset.Name, Value: lset.Value})
```

Comment:
Before this change we need to adjust Querier logic: https://github.com/thanos-io/thanos/blob/865d5ec710b5651ead08a7dc73864a6f5dbeeaa9/pkg/query/storeset.go#L248

Otherwise, sidecar + store gateway will be assumed a "duplicate" in Querier. Let's think about how we can fix this.

The storeAPI duplication detection logic is embedded in Querier to make sure user doesn't put many sidecars that upload blocks with the same external labels (common bug). I think we might want to change `storeset` behavior to look on StoreAPI  [`InfoResponse.StoreType`](https://github.com/thanos-io/thanos/blob/a460fa613d79877ee0b927650c9314cfb62fb010/pkg/store/storepb/rpc.pb.go#L34)  

I only don't like the fact that for Querier all of this is no longer `some StoreAPI` but actually `Sidecar, Store...` 

I might think we actually would need to change StoreType to `Producer, Browser` as that would be a good abstraction over different implementations. 

Thoughts @povilasv @GiedriusS @brancz ?

Reply:
Am I missing something or shouldn't this just be the set of all labelsets seen in all blocks?

Reply:
Can you elaborate?

Or better: Let's maybe quickly write up some proposal as we want consistent thing everywhere we do selectors + maybe selecting prefixes as well.

cc @jojohappy